js/models/entity.js

```diff
@@ -0,0 +1,49 @@
+class Entity{
+    constructor(imagePath, rotation, direction,width,height){
+        this.imagePath = imagePath;
+        this.rotation = rotation;
+        this.direction = direction;
+        this.width = width;
+        this.height = height;
+    }
+    
+    getImagePath(){
+        return this.imagePath;
+    }
+    
+    getRotation(){
+        return this.rotation;
+    }
+    
+    getDirection(){
+        return this.direction;
+    }
+    
+    getWidth(){
+        return this.width;
+    }
+    
+    getHeight(){
+        return this.height;
+    }
+    
+    setImagePath(imagePath){
+        this.imagePath = imagePath;
+    }
+    
+    setRotation(rotation){
+        this.rotation = rotation;
+    }
+    
+    setDirection(direction){
+        this.direction = direction;
+    }
+    
+    setWidth(width){
+        this.width = width;
+    }
+    
+    setHeight(height){
+        this.height = height;
+    }
+}
```

Comment:
If this is just a data class, can we forego the getters/setters? Someone more well versed in javascript standards should chime in, but this seems excessive

Reply:
At this time. I don't think the getters or setters are needed.

Reply:
@SpyMaster356 Just for my own knowledge, what is standard practice in js when it comes to getters/setters?

Reply:
From what I gathered on Mozilla's Javascript reference page, it adheres to generic object oriented standards, and since this is nothing more than a data-holding struct, we don't need to worry much about data hiding.

Fun fact: there's some syntax sugar for this
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

Reply:
Generally, there isn't much need for getters and setters, but there are use cases such as:
- When you want do extra things when setting a value
- When the value is computed from other properties
- When you want to restrict setting a value.